virtual_ecosystem/models/animal/cnp.py

```diff
@@ -0,0 +1,140 @@
+"""The :mod:`~virtual_ecosystem.models.animal.cnp` module contains the class for
+managing pools of stoichiometric explicit mass: carbon (C), nitrogen (N), and phosphorus
+(P).
+"""  # noqa: D205
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+
+@dataclass
+class CNP:
+    """A dataclass representing Carbon (C), Nitrogen (N), and Phosphorus (P) mass.
+
+    This class features common operations on CNP mass, including arithmetic
+    manipulations, stoichiometric calculations, and ratio/proportion retrieval.
+
+    Attributes:
+        carbon (float): The mass of carbon in the entity [kg].
+        nitrogen (float): The mass of nitrogen in the entity [kg].
+        phosphorus (float): The mass of phosphorus in the entity [kg].
+    """
+
+    carbon: float
+    nitrogen: float
+    phosphorus: float
+
+    @property
+    def total(self) -> float:
+        """Calculate the total combined mass of C, N, and P.
+
+        Returns:
+            float: The sum of carbon, nitrogen, and phosphorus mass.
+        """
+        return self.carbon + self.nitrogen + self.phosphorus
```

Comment:
See my comment about negative values. This will make no sense if any of the values is negative.